### ISSUE TITLE
Why custom directive not reflecting changes in its code immediately

### ISSUE THREAD

**Issue by hhsadiq** [laravel/framework#9732]
I am writing a simple custom directive in Laravel. Whenever I make some changes in the code of custom directive, its not reflected in the view until I

 - Comment the directive in view.
 - Reload the page
 - Uncomment the directive
 - Reload the page to finally get the changes

**Custom directive code in global.php**

    Blade::extend(function($value, $compiler)
    {
    	$pattern = $compiler->createMatcher('molvi');
    	return preg_replace($pattern, '$1<?php echo ucwords($2); ?>', $value);
    });

**Directive call in view**

    @molvi('haji') //this will output 'Haji' due to ucwords($2)

    //the ucwords() is replaced with strtolower()
    @molvi('haji') //this will still output 'Haji'

I am converting the words to uppercase. When lets say I want to use `strtolower()` instead of `ucwords()`, I have to repeat above steps to get changes reflected.

**UPDATE**

I have tried to clear the cache with various methods as described in [this thread](http://stackoverflow.com/questions/16971445/how-i-can-disable-templates-caching-in-development-mode) but still no success.

**Comment by lukasgeiter** [comment]
The problem is that the compiled views are cached and you can't disable that. You can however clear the files. Either manually by deleting everything in `storage/framework/views` or by running the command `php artisan view:clear`

**Comment by hhsadiq** [comment]
@lukasgeiter thanks for your reply. I have got your point. As far as `php artisan view:clear` is concerned, there is no such command. Running this command gives this error.
```
  [InvalidArgumentException]            
  Command "view:clear" is not defined.  
  Did you mean one of these?            
      cache:clear                       
      auth:clear-reminders              
      view:publish                      
```

The view has only this command

    view
       view:publish                 Publish a package's views to the application

**Comment by lukasgeiter** [comment]
The command is relatively new, you probably need to update the framework.

**Comment by hhsadiq** [comment]
Ok right. I am using laravel 4.2.1. This command is in laravel 5.0/5.1......right...?

**Comment by lukasgeiter** [comment]
Yes it is only available in 5.1. You can write your own command though.
Here's the source of the one from 5.1: [ViewClearCommand](https://github.com/laravel/framework/blob/5.1/src/Illuminate/Foundation/Console/ViewClearCommand.php)

**Comment by hhsadiq** [comment]
Thanks for your help @lukasgeiter  . The StackOverFlow usually proves not to be helpful in more serious questions. Again thanks.
